### DISCUSSION TITLE
MetalsResetWorkspace command doesn't seem to do anything (+ How do I "un-import" a sbt workspace?)

### DISCUSSION THREAD

**Discussion by pepper-lees** [scalameta/nvim-metals#582] [Q&A] [answered]
Hi,

I wanted to "un-import" build.sbt from one of my project submodules, because its dependencies are declared inside the _root_ project's build.sbt. And since I had Metals import build.sbt from the submodule, it seems to not recognize its dependencies, so all of the imports are being marked as not recognized. For example:

```
x import org.chipsalliance.cde.config.{Config}     ■ object chipsalliance is not a member of package org
```

I tried running `:MetalsResetWorkspace` to make it "forget" everything and start fresh, but when I run the command, nothing seems to happen. Is there something else I can do to reset it?

**Comment by ckipp01** [comment, marked as answer]
In this scenario if the wrong root was originally detected you'll want to just blow away the `.metals/`, `.bloop/`, and `.bsp/` directories that were created in the incorrect way. Then re-open the project from the root `build.sbt` and let it re-detect it. nvim-metals has some logic to traverse down and find the root `build.sbt`, but if there are situations where there are directories inbetween or something like that, then it might have issues.

**Reply by pepper-lees** [reply]
Ok, thanks!
